File: 532-k-diff-pairs-in-an-array/532-k-diff-pairs-in-an-array.py
class Solution:
    def findPairs(self, nums: List[int], k: int) -> int:
      c = collections.Counter(nums)
      res = 0
      for i in c:
        if (k==0 and c[i]>1) or (k!=0 and c[i+k] > 0):
          res+=1
      return res

File: 532-k-diff-pairs-in-an-array/532-k-diff-pairs-in-an-array.cpp
class Solution {
public:
  int findPairs(vector<int>& nums, int k) {
      unordered_multiset<int> ums(nums.begin(), nums.end());
      unordered_set<int> us(nums.begin(), nums.end());
      int ret = 0;
      for (int i : us) if (ums.count(i + k) > !k) 
        ++ret;
      return ret;
  }
};